### ISSUE TITLE
Cache notes not available

### ISSUE THREAD

**Issue by Sajomagc** [cgeo/cgeo#11232]
c:geo 2021.07.10-RC
Samsung S10  Android 11

When opening a geocache in c:geo the cache notes from the geocaching.com website do not synchronize. 
Updating the geocache in c:geo and restart of the mobile phone did not solve the problem.
I always have the solutions for my mysteries in the cache notes. They are no longer available in c:geo. 
Synchronization of personal waypoints works fine, also from cache notes to c:geo.

## System information 

```
Paste your system information here
```

**Comment by Lineflyer** [comment]
Confirmed, but I guess its a Website change.. AFAIR I saw it working last days

**Comment by murggel** [comment]
Just looked at it - can confirm that's a change in the website - the regExpr has to be adapted...

I can look at it.

**Comment by murggel** [comment]
BTW - how about to move those regExpr into an own "service" to avoid the necessity of a release if those changes happen...

**Comment by murggel** [comment]
And do we need a "live"-integration-test so we can check against the current web-site?

**Comment by murggel** [comment]
@Lineflyer on which branch should I base this change?

**Comment by fm-sys** [comment]
Definitely release IMHO to get it out quickly ;-)

**Comment by murggel** [comment]
We don't have a release without beta?

**Comment by moving-bits** [comment]
> Definitely release IMHO to get it out quickly ;-)

Not sure, as this would mean we would have to release our current beta as official release - don't know if we are already willing to do so.
But AFAIK @Lineflyer has put the old state of `release` branch (without our UI changes) aside under a different name.

**Comment by bekuno** [comment]
> 
> But AFAIK @Lineflyer has put the old state of `release` branch (without our UI changes) aside under a different name.

This is the branch `preUI_release`.
Maybe @Lineflyer can create a CI configuration as a first step, so we can develop in direction of both *release branches.

**Comment by fm-sys** [comment]
>But AFAIK @Lineflyer has put the old state of release branch (without our UI changes) aside under a different name. 

Oh, OK. Didn't know that. 

But be aware that this would AFAIK also cause that the current beta users go back to the old release which could probably have some unwanted side effects and we will receive much negative feedback why we have removed the new filter system again... Only way to avoid this would be using an old date as version ID (before current beta). But I'm not sure if we can manually edit this in our CI for release builds.

OTOH, the change is not _that_ critical, so it might be ok to wait some weeks until beta is finally ready. But I cannot really estimate how long the beta will still take...

**Comment by Lineflyer** [comment]
Thats true....we should IMHO only go back to the preUI_release in case of emergency (e.g. login problems).
Seems the fix has to wait a bit to be available to all.

**Comment by bekuno** [comment]
Then the PR should go to `preUI_release` and merged to `release*  and from `release` to `master`.
If we need a hotfix release then we should release this and in parallel a new beta release.
This needs a good timing ...

**Comment by Lineflyer** [comment]
That could work...the only problem is, that our RC/beta version code is always one day back....so we need to manipulate it a bit or just produce a release asap and wait until we can produce a RC/beta with versionCode > versionCode of normal release.

**Comment by Lineflyer** [comment]
As soon as the fix is merged to `preUI_release` please run this CI job:
https://ci.cgeo.org/view/Release_Nightly/job/cgeo%20release%20preUI%20(draft)/

If I configured it correctly it should build an APK with correct signing and upload it to https://download.cgeo.org/cgeo-release-preUI.apk

Do this on the same day of merge, so we have a ready version with that date stored.
Afterwards we can test and the next/overnext day try to release preUI + beta in parallel---not sue if that works smoothly.

**Comment by murggel** [comment]
PR #11239 is ready for review.

**Comment by baiti** [comment]
> BTW - how about to move those regExpr into an own "service" to avoid the necessity of a release if those changes happen...

With "service", what exactly do you mean? I believe you're talking about instantiating a server component that would act as a proxy between geocaching.com website and cgeo? Changes to the UI would be handled in that proxy, the API between cgeo and the proxy would never change, consequently changes in the proxy would not require rebuilding / testing / releasing a "fix" to all the users? Is it that, what you're suggesting?

**Comment by murggel** [comment]
> > BTW - how about to move those regExpr into an own "service" to avoid the necessity of a release if those changes happen...
> 
> With "service", what exactly do you mean? I believe you're talking about instantiating a server component that would act as a proxy between geocaching.com website and cgeo? Changes to the UI would be handled in that proxy, the API between cgeo and the proxy would never change, consequently changes in the proxy would not require rebuilding / testing / releasing a "fix" to all the users? Is it that, what you're suggesting?

Correct. I created issue #11235 for further discussion

**Comment by Lineflyer** [comment]
Merged to `preUI_release` and upstream over `release` to `master`.
Thanks @murggel

**Comment by Lineflyer** [comment]
I am sorry to say, but the fix seems either not to work or something changed again.
I tried it several times with current nighly (which includes this fix), but no personal note parsed for me from the server. The other direction (upload note with c:geo) is OK.

**Comment by Lineflyer** [comment]
@murggel Could you again take a look. I stopped preparing the release for now as the fix jsut does not work for me.

**Comment by murggel** [comment]
> @murggel Could you again take a look. I stopped preparing the release for now as the fix jsut does not work for me.

Ok, will have a look

**Comment by murggel** [comment]
> 
> Smoke test failed....the personal note is not working for me.
> I reopened #11232

With preUI_release it works in my emulator.

The code-snippet on the web-page is 
`<div id="srOnlyCacheNote" class="visually-hidden">personal test note
with linebreak</div>`
and the regExpr is `<div id=\"srOnlyCacheNote\"[^>]*>(.*?)</div>`

Probably a `>` in the note can lead to a problem. But I guess in your example this is not the case...
Edit: Hm...but the `^>` is in the tag and not the text-element, so no problem in my testcase

Also linebreaks are recognized in my test correctly. 
So I have no idea, why this is not working...

But it seems, that the change is only contained in the preUI_release and not in release or master as you mentioned above... 
But also I guess, you checked it with preUI_release.apk...

I am totally confused ...

_Originally posted by @murggel in https://github.com/cgeo/cgeo/issues/11242#issuecomment-881922559_

**Comment by fm-sys** [comment]
Indeed, commit 0e09de232aef9f4c250a0d1e23d147b069fe9558 still needs to be cherry-picked to release and then merged back to master...

**Comment by Lineflyer** [comment]
> Indeed, commit 0e09de2 still needs to be cherry-picked to release and then merged back to master...

**Comment by Lineflyer** [comment]
Indeed I used my `origin` remote instead of the `upstream` remote.

@murggel Sorry for wasting your time. I should know git much better, maybe it was the short night after watching the rising water flowing around my ICE train for 4 hours. :(

**Comment by murggel** [comment]
> Indeed I used my `origin` remote instead of the `upstream` remote.
> 
> @murggel Sorry for wasting your time. I should know git much better, maybe it was the short night after watching the rising water flowing around my ICE train for 4 hours. :(

@Lineflyer no problem, only I doubted on myself...
Hope your region is not affected from the water

**Comment by MagpieFourtyTwo** [comment]
This very minute I wanted to submit an issue because PNs are not sync'ed anymore ... searched in the issues for a possibly already existing one ... found none. While starting to write, I recognized the "new version is out" banner, so, to be up to date (actually, to be able to give the latest version number ;) ), I just quickly installed this latest version, tried again ... and \*plopp* PN sync works again! 

Checking the commits, I stumbled over [0ac91b4](https://github.com/cgeo/cgeo/commit/0ac91b4934e4612ad8a89c4028966b7a806f6695), landed right here ... and could hardly believe my eyes ... issue not even written, problem fixed.

Wow - this was a *really* quick one!  ;))

Anyway - I just wanted to express my great gratitude for that you are so tirelessly committed to c:geo and usually respond (and act) promptly. Every time anew, I am thrilled. BIG THANKs to all involved!
